src/2a_analyze_MPA.jl

```diff
@@ -36,6 +36,60 @@ include("common.jl")
         return nothing
     end
 
+        """
+        filter_distances(
+            target_rast::Raster,
+            gdf::DataFrame,
+            dist_nm
+        )::Raster
+
+    Filter pixels from target_raster that are within `dist_nm` (nautical miles) from a geometry
+    in gdf. Target_raster and gdf should be in the same CRS (EPSG7844:GDA2020 for GBR-reef-guidance-assessment).
+
+    # Arguments
+    - `target_rast` : Raster of suitable pixels (Bool) to filter pixels from.
+    - `gdf` : DataFrame with `geometry` column that contains vector objects of interest.
+    - `dist_nm` : Filtering distance from geometry object in nautical miles.
+
+    # Returns
+    - `tmp_areas` : Raster of filtered pixels containing only pixels within target distance from
+    gdf geometries.
+    """
+    function filter_distances(target_rast::Raster, gdf::DataFrame, dist_nm)::Raster
+        tmp_areas = copy(target_rast)
+        raster_lon = Vector{Float64}(tmp_areas.dims[1].val)
+        raster_lat = Vector{Float64}(tmp_areas.dims[2].val)
+
+        @floop for (lon_ind, lon) in enumerate(raster_lon)
+            for (lat_ind, lat) in enumerate(raster_lat)
+                if tmp_areas.data[lon_ind, lat_ind]
+
+                    point = AG.createpoint()
+                    AG.addpoint!(point, lon, lat)
+
+                    pixel_dists = AG.distance.([point], Ports.geometry)
+                    geom_point = gdf[argmin(pixel_dists), :geometry]
+                    geom_point = (AG.getx(geom_point, 0), AG.gety(geom_point, 0))
+
+                    dist_nearest = Distances.haversine(geom_point, (lon, lat))
+                    dist_nearest = dist_nearest / 1852
+
+                    if dist_nearest < dist_nm
+                        tmp_areas.data[lon_ind, lat_ind] = 1
+                    else
+                        tmp_areas.data[lon_ind, lat_ind] = 0
+                    end
+                end
+            end
+        end
+
+        return tmp_areas
+    end
```

Comment:
I'm not sure this function is really necessary? There should be similar functionality available in the geospatial packages

Reply:
Ignore this comment for now, working with polygons are not as straightforward as I initially thought but I can take care of it when the time comes.